extending/python_requires.rst

```diff
@@ -1,160 +1,245 @@
 .. _python_requires:
 
-Python requires: reusing python code in recipes [EXPERIMENTAL]
-==============================================================
+Python requires: reusing code [EXPERIMENTAL]
+============================================
 
 .. warning::
 
     This is an **experimental** feature subject to breaking changes in future releases.
 
-The ``python_requires()`` feature allows to extend the recipes reusing Python code from other **conanfile.py** easily, even for inheritance
-approaches. The code to be reused will be in a *conanfile.py* recipe, and will be managed as any other Conan package.
+The ``python_requires()`` feature is a very convenient way to share files and code between
+different recipes. A *Python Requires* is just like any another recipe, it is the way it is
+required from the consumer what makes the difference.
 
-Let's create for example some reusable base class:
+The *Python Requires* recipe file, besides exporting its own required sources, can export
+files to be used by the consumer recipes and also python code in the recipe file itself.
 
-.. code-block:: python
+Let's have a look at an example showing all its capabilities (you can find all
+the sources in ....):
 
-    from conans import ConanFile
+ - Python requires recipe:
 
-    class MyBase(ConanFile):
-        def source(self):
-            self.output.info("My cool source!")
-        def build(self):
-            self.output.info("My cool build!")
-        def package(self):
-            self.output.info("My cool package!")
-        def package_info(self):
-            self.output.info("My cool package_info!")
+    .. code-block:: python
 
-With this conanfile, we can export it to the local cache to make it available, and also upload to our remote:
+        import os
+        import shutil
+        from conans import ConanFile, CMake, tools
+        from scm_utils import get_version
 
-.. code-block:: bash
 
-    $ conan export . MyBase/0.1@user/channel
-    $ conan upload MyBase/0.1@user/channel -r=myremote
+        class PythonRequires(ConanFile):
+            name = "pyreq"
+            version = "version"
 
-It is not necessary to "create" any package binaries, or to ``upload --all``, because there are no binaries for this recipe.
+            exports = "scm_utils.py"
+            exports_sources = "CMakeLists.txt"
 
-Now, using the ``python_requires()`` we can write a new package recipe like:
 
-.. code-block:: python
+        def get_conanfile():
```

Comment:
I would like to have this confirmed as the canonical/recommended way of using python requires

Reply:
Yes, this is something we have to agree, @conan-io/barbarians